parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReaderForDouble.java

```diff
@@ -0,0 +1,53 @@
+/* 
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.bytestreamsplit;
+
+import org.apache.parquet.io.ParquetDecodingException;
+
+import java.io.IOException;
+
+public class ByteStreamSplitValuesReaderForDouble extends ByteStreamSplitValuesReader {
+
+  private static final int DOUBLE_SIZE_IN_BYTES = 8;
+  private byte[] valueByteBuffer;
+
+  public ByteStreamSplitValuesReaderForDouble() {
+    super(ByteStreamSplitValuesReaderForDouble.DOUBLE_SIZE_IN_BYTES);
+    valueByteBuffer = new byte[DOUBLE_SIZE_IN_BYTES];
+  }
+
+  @Override
+  public double readDouble() {
+    try {
+      gatherElementDataFromStreams(valueByteBuffer);
+      long b1 = valueByteBuffer[0] & 0xFF;
+      long b2 = valueByteBuffer[1] & 0xFF;
+      long b3 = valueByteBuffer[2] & 0xFF;
+      long b4 = valueByteBuffer[3] & 0xFF;
+      long b5 = valueByteBuffer[4] & 0xFF;
+      long b6 = valueByteBuffer[5] & 0xFF;
+      long b7 = valueByteBuffer[6] & 0xFF;
+      long b8 = valueByteBuffer[7] & 0xFF;
+      long value = b1 | (b2 << 8) | (b3 << 16) | (b4 << 24) | (b5 << 32) | (b6 << 40) | (b7 << 48) | (b8 << 56);
+      return Double.longBitsToDouble(value);
+    } catch (IOException | ArrayIndexOutOfBoundsException ex) {
+      throw new ParquetDecodingException("Could not read double.");
```

Comment:
It would be hard to debug the error especially when it throws in production env, when we just show a simple string "Could not read double.". I suggest to include the 'ex' itself and wrap it up with ParquetDecodingException so that the exception can have more information. It is also recommended to print out more information of the data stream itself so that it is easy to troubleshoot the error.

Reply:
Passed the exception object to ParquetDecodingException